warehouse/query-core/src/main/java/datawave/query/planner/FederatedQueryPlanner.java

```diff
@@ -0,0 +1,100 @@
+package datawave.query.planner;
+
+import datawave.query.CloseableIterable;
+import datawave.query.config.ShardQueryConfiguration;
+import datawave.query.exceptions.DatawaveQueryException;
+import datawave.query.planner.pushdown.rules.PushDownRule;
+import datawave.query.tables.CountingShardQueryLogic;
+import datawave.query.tables.ScannerFactory;
+import datawave.query.tables.ShardQueryLogic;
+import datawave.query.util.MetadataHelper;
+import datawave.webservice.common.logging.ThreadConfigurableLogger;
+import datawave.webservice.query.Query;
+import datawave.webservice.query.configuration.GenericQueryConfiguration;
+import datawave.webservice.query.configuration.QueryData;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+
+import java.util.Collection;
+
+public class FederatedQueryPlanner extends QueryPlanner implements Cloneable {
+    
+    QueryPlanner originalQueryPlanner;
```

Comment:
There are going to be more than once instance of a query planner based on the query holes